NFIQ2/NFIQ2Algorithm/src/tool/nfiq2_ui_utils.cpp

```diff
@@ -235,7 +235,7 @@ NFIQ2UI::checkThreads(const std::string &optarg)
 }
 
 std::string
-NFIQ2UI::formatDouble(const double &d, const int precision)
+NFIQ2UI::formatDouble(const double &d, const long precision)
```

Comment:
`long` is still signed, thus the `static_cast` required on 272. This should be unsigned, and thus impossible to set an invalid (negative) precision. Maybe this should be a `std::string::size_type` or simply `uint8_t` (unless you're expecting precision >255).